hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapReplicationStateHolder.java

```diff
@@ -96,7 +96,7 @@
     // operations, which meant that the index did not include some data.
     protected transient List<MapIndexInfo> mapIndexInfos;
 
-    private transient Address target;
+    private final transient ThreadLocal<Address> target = new ThreadLocal<Address>();
```

Comment:
You want a  ThreadLocal instance per MapReplicationStateHolder? Normally ThreadLocals are static.

Reply:
good catch! fixed